### ISSUE TITLE
The addition of a AWS component for locomotives:

### ISSUE THREAD

**Issue by alexkubel** [TeamOpenIndustry/ImmersiveRailroading#1321]
by making AWS a discrete component (as opposed to a widget) it can be more easily operated by a augment to trip it and can have it's own keybind (potentially), aswell as being able to be sensed to trip other parts of Automated Train Control for when the player is not observant.
further you could have a JSON line that specifies if a locomotive has AWS installed or not.

**Comment by bakinser** [comment]
This could be made more generically inclusive by calling it an In-Cab Signalling system. I believe the German/Austrian PZB system functions similarly. I assume due to the lack of internal support for signals in IR, that whether it trips the system is based on the redstone input of the augment?

**Comment by alexkubel** [comment]
yeah that's the idea by having it able to interact with redstone, you could have it control things agnostically (e.g.) with redstone lamp signals or whatever signalling mod you prefer.

I chose AWS as it's the british system, PZB is not a system I am directly familiar with. it will need a standardised name in the mod, so something generic could probably be found to cover them collectively.
